src/riscv/lib/src/pvm/tezos.rs

```diff
@@ -43,10 +45,61 @@ use crate::machine_state::registers::a1;
 use crate::machine_state::registers::a2;
 use crate::machine_state::registers::a3;
 use crate::machine_state::registers::a6;
+use crate::pvm::linux::parameters::ECALL_WIDTH;
 use crate::state_backend::Cell;
 use crate::state_backend::ManagerReadWrite;
 use crate::state_backend::ManagerWrite;
 
+/// Status of a Tezos host function call
+enum CallStatus {
+    /// The call is suspended and will be resumed later.
+    Suspended,
+
+    /// The host function handler has finished successfully.
+    Success(XValue),
+
+    /// The host function handler errored.
+    Error(SbiError),
+}
+
+/// The result of handling a host function.
+/// This is similar to [`crate::pvm::linux::parameters::SystemCallResultExecution`] but there
+/// are slight differences between what we allow tezos and linux calls to do.
+struct TezosCallResult {
+    /// On success, the value is written to output register `a0`.
+    status: CallStatus,
+
+    /// Usually, handling a host function involves an increment to the pc.
+    pc_update: ProgramCounterUpdate<Address>,
+}
+
+impl TezosCallResult {
+    const SUSPENDED: Self = TezosCallResult {
+        status: CallStatus::Suspended,
+        pc_update: ProgramCounterUpdate::Next(ECALL_WIDTH),
+    };
+
+    /// The error is handled by the caller (kernel)
+    /// We move to the next instr (increment the pc)
+    fn error(err: SbiError) -> Self {
+        Self {
+            status: CallStatus::Error(err),
+            pc_update: ProgramCounterUpdate::Next(ECALL_WIDTH),
+        }
+    }
+
+    /// Usually the PC is incremented after handling an ECALL
```

Comment:
This comment should say what the function does rather than when it would be used. What do you think about "Advance the pc to the next instruction"